tools/clang/test/HLSLFileCheck/hlsl/operators/ternary-enum.hlsl

```diff
@@ -0,0 +1,36 @@
+// RUN: %dxc -ast-dump -T vs_6_0 %s | FileCheck %s
+// RUN: %dxc -ast-dump -T vs_6_0 -HV 2021 %s | FileCheck %s
+// RUN: %dxc -ast-dump -T vs_6_0 -DCLASS %s | FileCheck %s
+// RUN: %dxc -ast-dump -T vs_6_0 -DCLASS -HV 2021 %s | FileCheck %s
+
+// CHECK: VarDecl
+// CHECK-SAME: referenced g_c 'const MyEnum' static cinit
+// CHECK-NEXT: ConditionalOperator
+// CHECK-SAME: 'MyEnum'
+// CHECK-NEXT: CXXBoolLiteralExpr
+// CHECK-SAME: 'bool' true
+// CHECK-NEXT: DeclRefExpr
+// CHECK-SAME: 'MyEnum' EnumConstant
+// CHECK-SAME: 'One' 'MyEnum'
+// CHECK-NEXT: DeclRefExpr
+// CHECK-SAME: 'MyEnum' EnumConstant
+// CHECK-SAME: 'Two' 'MyEnum'
+
+enum
+#ifdef CLASS
+  class
+#endif
+MyEnum {
+    One,
+    Two,
+};
+
+#ifdef CLASS
+static const MyEnum g_c = true ? MyEnum::One : MyEnum::Two;
+#else
+static const MyEnum g_c = true ? One : Two;
```

Comment:
Does it have to be static const? Does it have to be the initialization? Just wondering how narrow this is. Clearly we aren't testing the different things we can plug into the ternary operator (or select()) as well as we should. What do you think about expanding this to include a number of other types? Maybe not in this change, but I think it's a test we need.